### PR TITLE
Implement the remove signal game command

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Pretty simple one v. similar to create but with some easier logic.

### CODE REVIEW

File: src/OpenLoco/src/GameCommands/Track/RemoveSignal.cpp [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+#include "Map/TileManager.h"
+#include "Map/Track/TrackData.h"
+#include "Map/TrackElement.h"
+#include "Objects/TrainSignalObject.h"
+#include "Vehicles/Vehicle.h"
+#include "ViewportManager.h"
+
+namespace OpenLoco::GameCommands
+{
+    static World::TrackElement* getElTrackAt(const SignalRemovalArgs& args, const World::Pos3 pos, const uint8_t index)
```
Comment (ZehMatt) [inline src/OpenLoco/src/GameCommands/Track/RemoveSignal.cpp:15]:
```suggestion
    static World::TrackElement* getElTrackAt(const SignalRemovalArgs& args, const World::Pos3 pos, const uint8_t sequenceIndex)
```
I think index is a bit too vague here.

**Review by ZehMatt — APPROVED** [review]
Code looks fine, I haven't tested this.

**Review by AaronVanGeffen — COMMENTED** [review]
I'm having trouble with signal ghost removal. Just hovering over a track with the signal placement tool may leave a ghost behind, though not always?? Seems to have to do with two-way/double signal placement.

**Comment by duncanspumpkin** [comment]
> I'm having trouble with signal ghost removal. Just hovering over a track with the signal placement tool may leave a ghost behind, though not always?? Seems to trigger more easily with two-way/double signal placement, but happens with single signal placement as well. Perhaps this is a handedness issue, i.e. only one side of the track is problematic?

found the issue it will cause mistakes on multi tile tracks. Fix coming soon.

**Review by AaronVanGeffen — APPROVED** [review]
Works fine now. Nice work!

**Pull request merged**